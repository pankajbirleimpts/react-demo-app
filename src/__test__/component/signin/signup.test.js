import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import Signup, { UnConnectedSignup } from '../../../components/signin/signup';
import { storeFactory, findByAttr } from '../../test/TestUtils';



const setup = (props) => {
    const store = storeFactory(props);
    const wrapper = shallow(
        <Signup store={store} />,
    )
        .dive()
        .dive();
    return wrapper;
};

describe('Signup component', () => {
    //   test('Snapshot of Signup component', () => {
    //     const props = {
    //       user: {
    //         isAuthenticated: false,
    //       },
    //     };
    //     const wrapper = setup(props);
    //     expect(wrapper).toMatchSnapshot();
    //   });


    test('Check the isAuthenticated in props', () => {
        const props = {
            user: {
                isAuthenticated: false,
            },
        };
        const wrapper = setup(props);
        const componentProps = wrapper.instance().props;
        expect(componentProps.user.isAuthenticated).toBe(props.user.isAuthenticated);
    });

    test('Check the Signup action in props', () => {
        const props = {
            user: {
                isAuthenticated: false,
            },
        };
        const wrapper = setup(props);
        const componentProps = wrapper.instance().props;
        expect(componentProps.signup).toBeInstanceOf(Function);
    });

    test('Checking redirect to dashboard if user is loggedin ', () => {
        const historyMock = { replace: jest.fn() };
        const props = {
            user: {
                isAuthenticated: true,
            },
            history: historyMock,
        };
        const wrapper = shallow(<UnConnectedSignup {...props} />);
        expect(historyMock.replace).toHaveBeenCalled();
    });

    test('Checking the form values ', async () => {
        const historyMock = { replace: jest.fn() };
        const singupMock = jest.fn();
        const props = {
            user: {
                isAuthenticated: false,
            },
            history: historyMock,
            signup: singupMock,
        };
        const formValues = {
            email: 'test@test.com',
            password: 'test@123',
            firstName: "Test",
            lastName: "User",
            balance: 0,
            role: "EMPLOYEE",
            country: "Indore",
            employeeId: "Test123"
        };

        const wrapper = mount(<UnConnectedSignup {...props} />);

        const emailInput = findByAttr(wrapper, 'email-input').find('input').first();
        const passwordInput = findByAttr(wrapper, 'password-input').find('input').first();
        const confirmPasswordInput = findByAttr(wrapper, 'confirmPassword-input').find('input').first();
        const firstNameInput = findByAttr(wrapper, 'firstName-input').find('input').first();
        const lastNameInput = findByAttr(wrapper, 'lastName-input').find('input').first();
        const employeeIdInput = findByAttr(wrapper, 'employeedId-input').find('input').first();
        const countryInput = findByAttr(wrapper, 'country-input').find('select').first();

        await act(async () => {
            emailInput.simulate('change', { persist: () => { }, target: { name: 'email', value: formValues.email } });
        });

        await act(async () => {
            passwordInput.simulate('change', { persist: () => { }, target: { name: 'password', value: formValues.password } });
        });

        await act(async () => {
            confirmPasswordInput.simulate('change', { persist: () => { }, target: { name: 'confirmPassword', value: formValues.password } });
        });

        await act(async () => {
            firstNameInput.simulate('change', { persist: () => { }, target: { name: 'firstName', value: formValues.firstName } });
        });

        await act(async () => {
            lastNameInput.simulate('change', { persist: () => { }, target: { name: 'lastName', value: formValues.lastName } });
        });

        await act(async () => {
            employeeIdInput.simulate('change', { persist: () => { }, target: { name: 'employeeId', value: formValues.employeeId } });
        });

        await act(async () => {
            countryInput.simulate('change', { persist: () => { }, target: { name: 'country', value: formValues.country } });
        });

        await act(async () => {
            wrapper.find('form').simulate('submit');
        });

        expect(singupMock).toHaveBeenCalledWith(formValues, expect.anything());
    });

});