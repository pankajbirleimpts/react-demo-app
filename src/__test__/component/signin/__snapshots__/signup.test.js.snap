// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component Snapshot of Signup component 1`] = `
<Grid>
  <Loading />
  <GridRow
    centered={true}
  >
    <GridColumn
      width="8"
    >
      <h2>
        Signup
      </h2>
      <Formik
        initialValues={
          Object {
            "confirmPassword": "",
            "country": "",
            "email": "",
            "employeeId": "",
            "firstName": "",
            "lastName": "",
            "password": "",
          }
        }
        onSubmit={[Function]}
        validationSchema={
          ObjectSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_defaultDefault": [Function],
            "_deps": Array [],
            "_excludedEdges": Array [],
            "_exclusive": Object {},
            "_mutate": undefined,
            "_nodes": Array [
              "country",
              "email",
              "confirmPassword",
              "password",
              "lastName",
              "firstName",
            ],
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "object",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "fields": Object {
              "confirmPassword": SchemaType {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "equalTo": false,
                },
                "_mutate": undefined,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "mixed",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                ],
                "transforms": Array [],
              },
              "country": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "required": true,
                },
                "_mutate": undefined,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
              "email": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "required": true,
                  "undefined": false,
                },
                "_mutate": undefined,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
              "firstName": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "max": true,
                  "required": true,
                },
                "_mutate": undefined,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
              "lastName": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "max": true,
                  "required": true,
                },
                "_mutate": undefined,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
              "password": StringSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_deps": Array [],
                "_exclusive": Object {
                  "min": true,
                  "required": true,
                },
                "_mutate": undefined,
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "string",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "tests": Array [
                  [Function],
                  [Function],
                ],
                "transforms": Array [
                  [Function],
                ],
              },
            },
            "tests": Array [],
            "transforms": Array [
              [Function],
            ],
          }
        }
      >
        <Component />
      </Formik>
    </GridColumn>
  </GridRow>
</Grid>
`;
